### PR TITLE
Map monitoring ping

### PR DESCRIPTION
**Pull request by darkdrgn2k**
Add secondary dynamic json file from a server that periodically pings live ipv6 addresses in node repositorty to update the map with ACTUAL up/down status of nodes

### CODE REVIEW

**Comment by garrying** [comment]
@darkdrgn2k Have you been testing this on `localhost`? Seeing a `Access-Control-Allow-Origin` error. Also, we should look at using a different JS pattern for fetching and rendering from the endpoint; Promises, or success handlers so that data is rendered when it is actually retrieved. The bottom part of http://api.jquery.com/jQuery.getJSON/

**Comment by darkdrgn2k** [comment]
This code has been sitting on a shelf for a while now, i remember solving the CROS at one point but maybe i didn't commit or something, I'm not sure, or maybe it was the git stuff i was dealing with. Regardless thanks for testing!.

@garrying I updated the remote side to ping.json.php which throws up some CROS headers to the json file to correct the problem. I have not tested it but it SHOULD work.

**Comment by darkdrgn2k** [comment]
ASYNC json ajax download error handling race condition...

@garrying Did the map LOAD even though there was a CROS error? or was the error blocking?

The main data from git hub renders upon successful download.

The new auxiliary monitoring data is currently not REQUIRED to render the map, and yes has no error checks at all, and is based on a "I hope it has it by the time i need it" philosophy. 

Anyway, Not the best solution and probably one of the reasons I didn't do a pull request earlier, but to be honest I don't know how to fix this problem without doing some MAJOR over hauling and i don't think its imperative at this time. So instead IF the data is available when it needs it, it will be used, otherwise its just skipped and not processed this time. A failure condition would probably not change anything at this time.

If anyone has a better idea of how to deal with this, go to town, but the code has been sitting on a shelf for at least a month now so i figured id throw it into the wild and see what happens. Or throw it into an issue or whatever

I just didn't want this sitting on a shelf any more

File: js/map.js [hunk 1]
```diff
@@ -3,9 +3,15 @@ var infowindow = null;
 var markers = [];
 var currentNodeListURL;
 var circle = null;
+var pingData;
 
 function initialize() {
-	
+
+	//Pull Ping Data
+	$.getJSON("https://node2.e-mesh.net/ping.json.php", function (data) {
```
Comment (benhylau) [inline js/map.js:11]: Can we run this local?

**Comment by darkdrgn2k** [comment]
Sure as long as you can run a PHP script as a cron on the server..

File: js/map.js (hunk 1 above)
Comment (darkdrgn2k) [reply js/map.js:11]: Sure as long as you can run a PHP script as a cron on the server..
https://github.com/darkdrgn2k/ToMeshStuff/blob/master/cjdnsPING.php

Local will also need access to hyperborea. 

Other option is to have a remote cjdns node upload the data to the server very run.

Comment (darkdrgn2k) [reply js/map.js:11]: Also another php script that runs on a hyperborea node that is "stage 2" of this is the linking script.

https://github.com/tomeshnet/tomesh.net/tree/MapMonitoring-Links

https://github.com/darkdrgn2k/ToMeshStuff/blob/master/cjdnsDump.php

**Comment by garrying** [comment]
@darkdrgn2k Sorry for not following up sooner. I had another, higher-level, thought on this: we should address the ambiguity of the external resource. Maybe we can provide documentation, or comments, around what is going on with it and how to replicate the resource locally.

If there's something new in the works, feel free to close.

**Comment by benhylau** [comment]
@darkdrgn2k and I were talking about how the website is turning complex enough that it will be a lot of manual steps to set up. Perhaps it is better to make it into a docker container that does:

* The current jekyll static site + rebuild cron + rebuild hook + nginx + letsencrypt as [documented here](https://github.com/tomeshnet/documents/blob/master/service_setup/website.md)
* Running cjdns + serve website over the IPv6 with nginx configs
* Scripts that ping nodes over cjdns to feed the map

This way the set up is code automated and steps to run is basically to run a docker container providing a config file that holds tomesh-specific URLs and keys.

**Comment by darkdrgn2k** [comment]
There is also the question of grafana... If that's what we will be doing. And prometheus server too. 

The ping and crawl script are just cron jobs with a script so it not overly complicated. 

It writes a file that is read by the js I don't know how jakyll will handle that

**Comment by garrying** [comment]
Dockerizing the site makes sense. Maybe the map and monitoring could be split into its own thing. Let's close this for now.

**Comment by benhylau** [comment]
https://github.com/tomeshnet/monitoring

**Pull request closed without merging**